ui/app/components/ui/color-indicator/color-indicator.scss

```diff
@@ -0,0 +1,61 @@
+@use "utilities";
+@use "design-system";
+
+$sizes: (
+  'large': 6,
+  'medium': 5,
+  'small': 4,
+);
+
+.color-indicator {
+  $self: &;
+
+  border: 1px solid transparent;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  &__inner-circle {
+    background-color: transparent;
+  }
+
+  @each $variant, $size in $sizes {
+    &--size-#{$variant} {
+      height: #{2 * $size}px;
+      width: #{2 * $size}px;
+      border-radius: #{$size}px;
+
+      #{$self}__inner-circle {
+        border-radius: #{$size}px;
+        height: #{$size}px;
+        width: #{$size}px;
+      }
+
+      #{$self}__icon {
+        font-size: #{1.25 * $size}px;
+      }
+    }
+  }
+
+  @each $variant, $color in design-system.$color-map {
+    &--color-#{$variant} {
+      border-color: $color;
+      &#{$self}--partial-filled #{$self}__inner-circle {
+        background-color: $color;
+      }
+      &#{$self}--filled {
+        background-color: $color;
+      }
+      & #{$self}__icon {
+        color: #{utilities.choose-contrast-color($color)};
+      }
+    }
+  }
+
+  // separate iterator to ensure borderColor takes precedence
```

Comment:
When testing this locally in storybook, it seems that this `border-color` never actually has precedence. I'm just seeing the `color-indicator--color-{variant}` border color :thinking:

Reply:
Actually, no, it's only broken when I set the border colour to one of the `ERROR` colours.

Reply:
Oh I see - there's a mistake in the design system constants. I can fix that in a separate PR.

Reply:
https://github.com/MetaMask/metamask-extension/pull/10246